### DISCUSSION TITLE
School upload data mapping and its impact on course reports?

### DISCUSSION THREAD

**Discussion by thippeswamy123-ship-it** [Sunbird-Ed/Community#700] [General Discussion]
Hi @mohitga ,

We need your help to understand about course reports.

Bihar team has raised a request to upload school data to that state , but there are live improvement projects and courses are running now , there is a difference in the school code for many schools as compared to the existing data in DIKSHA PRODUCTION.

We want to understand, if we upload school data , will that impact course reports data? If yes, please let us know what kind of impact will it create in reports as well as in consumption of the course.

Please let us know if any additional information is needed from our end.

CC : @ayeshanaffia

**Comment by mohitga** [comment]
@amiableAnil @aks30 folks can you help @thippeswamy123-ship-it in the above ask.

**Reply by Jayaprakash8887** [reply]
There will be no impact related to consumption of a course or any other contents on the platform. However, with respect to reports, impact will be there on 
https://lern.sunbird.org/use/developer-guide/lms-service/reports/on-demand-exhaust/user-info-exhaust

Below reports related to Courses **wont** be impacted.
https://lern.sunbird.org/use/developer-guide/lms-service/reports/on-demand-exhaust/response-exhaust
https://lern.sunbird.org/use/developer-guide/lms-service/reports/on-demand-exhaust/progress-exhaust

**Comment by thippeswamy123-ship-it** [comment]
@AmiableAnil , can you please provide an update on this?

This is important for us to address a request from the state.

**Reply by Jayaprakash8887** [reply]
@thippeswamy123-ship-it We are not able to relate to "live improvement projects". What does this mean?

**Reply by thippeswamy123-ship-it** [reply]
@Jayaprakash8887  , i am referring to the project option under ML capabilities, however, we want to understand the impact on the course reports and consumption of the courses

**Reply by Jayaprakash8887** [reply]
Also, if the data update is only on school code of existing schools, we expect no impact on user-info-exhaust report as well. school code changes are key only in Single-signon (SSO) flow.

**Reply by thippeswamy123-ship-it** [reply]
Also, if the new upload of school data has cluster level changes what kind of impact on reports?, Is there any impact on the consumption of the courses?

 cc : @tb-thejaswini

**Reply by thippeswamy123-ship-it** [reply]
@Jayaprakash8887 

And also , there is a difference in the school code for many schools as compared to the existing data in diksha , will that impact reports/ course consumption?

**Reply by Jayaprakash8887** [reply]
@thippeswamy123-ship-it Reiterating again, as per the latest code and the documentation links shared above of SUNBIRD LERN, we do not see any impact on consumption of courses by users and impact maybe on one of the reports (user-info-exhaust). However, we are not sure about version and implementation in DIKSHA. Request you to kindly connect with them. 

Please Note: 
1. We suggest you to do the analysis first on a dev/pre-prod environment of DIKSHA to analyse any impact.
2. Update the existing school records using APIs instead of clearing the existing schools records and recreating with fresh data. Deleting existing schools may lead to greater impact.

**Reply by thippeswamy123-ship-it** [reply]
@Jayaprakash8887  , thanks for the clarification

**Reply by ayeshanaffia** [reply]
@Jayaprakash8887 While we check the version of DIKSHA, we would like to understand more on what kind of impact on one of the reports (user-info-exhaust) in SUNBIRD LERN as you mentioned above?

**Reply by ayeshanaffia** [reply]
@Jayaprakash8887 I hope you got a chance to look into this!

**Reply by Jayaprakash8887** [reply]
@ayeshanaffia No. Request you to do the changes in dev or pre-prod environment and analyse the same. We do not have scope for doing data impact analysis.

**Reply by ayeshanaffia** [reply]
@Jayaprakash8887 ok thanks!

**Comment by mohitga** [comment]
@aks30 can you help @thippeswamy123-ship-it  for ML related query

**Comment by HarishGangula** [comment]
As we there are no active updates, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
